buildroot-external/board/raspberrypi/rpi5-64/hassos-hook.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# shellcheck disable=SC2155
+
+function hassos_pre_image() {
+    local BOOT_DATA="$(path_boot_dir)"
+
+    mkdir -p "${BOOT_DATA}/slot-A/"
+    cp "${BINARIES_DIR}"/*.dtb "${BOOT_DATA}/slot-A/"
+    gzip --stdout "${BINARIES_DIR}"/Image > "${BOOT_DATA}/slot-A/kernel_2712.img"
+    cp "${BINARIES_DIR}"/*.dtbo "${BOOT_DATA}/slot-A/overlays/" 2>/dev/null || true
```

Comment:
Ok I've added now `BR2_PACKAGE_RPI_FIRMWARE=y` and an appropriate copy command.

Reply:
If we have to pick from those two options, then I vote for using self-built DTBs and overlays. We have to build and supply DTB for Yellow anyway and having the rest built from sources is IMHO more flexible and easier for maintenance than using those from the RPi repo.

Anyway it's been like this for ages and there's no urgency for changing that, but good point.

Reply:
> Anyway it's been like this for ages and there's no urgency for changing that, but good point.

Yeah changing for the other board is a different thing and doesn't belong into this PR.

However, for RPi5 we can build and deploy them from the Buildroot kernel build directly. I'll look int this.

Reply:
> We have to build and supply DTB for Yellow anyway and having the rest built from sources is IMHO more flexible and easier for maintenance than using those from the RPi repo.

So far we build and supply extra overlays through the `rpi-rf-mod` package, which is Homematic specific (speaking of which, maybe we should make that more clear in that package name :thinking: ). But other than that we do not supply overlays from the main Linux kernel source tree so far.

IMHO, the current situation is bad, as we mix the device tree from kernel source and overlays from the firmware.

Technically, the device tree is supposed to be provided by the machine's firmware. Like in the x86 world the ACPI tables are part of the BIOS. So there is definitely an argument to make that we should deploy device tree and kernel from the Raspberry Pi provided directory.

However, in the Arm world the boards are much more diverse, and quite often firmware (or part of the firmware) is bundled with the OS. That is also the case in Home Assistant OS: We deploy machine firmware along with the OS (and not just the device trees and overlay, but also Raspberry Pi's `start4.elf`, or U-Boot which on some boards can be seen as firmware). Now where the individual components of the "firmware" come from are almost implementation detail: The Raspberry Pi device trees and overlays are maintained in their Linux kernel source tree anyway. It is only a matter of do we deploy their pre-built one or do we deploy our own builds.

Deploying our own builds gives us more flexibility and proves to be less confusing/error-prone (as often the newer Linux kernel benefits from a newer/matching device tree).

So with this in mind, I agree let's deploy overlays from our build.

It seems that Buildroot doesn't support that use case yet. But it is a fairly simple extension for the Buildroot build system: https://github.com/home-assistant/buildroot/pull/18.

Reply:
> So far we build and supply extra overlays through the `rpi-rf-mod` package, which is Homematic specific (speaking of which, maybe we should make that more clear in that package name 🤔 ). But other than that we do not supply overlays from the main Linux kernel source tree so far.

Sorry, but didn't follow the discussions closely enough. So what is the exact issue with the `rpi-rf-mod` since I noticed that you removed its CONFIG options from the rpi5 config file? In fact, it perfectly builds and runs with a rpi5 and the device tree overlay is working nicely AFAICS.

> IMHO, the current situation is bad, as we mix the device tree from kernel source and overlays from the firmware.

I am not really sure if this is exactly the case for the RaspberryPi stuff. In fact, AFAIK even for RaspberryPiOS the makers are not directly using the in-tree device tree overlay stuff but rather uses the pre-build rpi-firmware dtbo files from their repository. So I am quite unsure if this would not construct an additional issue here if you guys would now switch to using the dtbo files from the kernel tree rather than pulling them from the rpi-firmware package.

And again. why is disabling/removing the rpi-rf-mod.dtbo required ATM?

Reply:
>  In fact, it perfectly builds and runs with a rpi5 and the device tree overlay is working nicely AFAICS.

Ok, I wasn't aware that this got tested already, so we can re-enable it if it's known to be working. However,  with just these two set:

```
BR2_PACKAGE_RPI_RF_MOD=y
BR2_PACKAGE_RPI_RF_MOD_DTS=y
```

I noticed that `rpi-rf-mod-rpi1.dtbo` gets deployed, which I think is just wrong for Raspberry Pi 5? This might be just a config issue, not sure.

Btw, you typically find the rationals behind a change in the individual commits (https://github.com/home-assistant/operating-system/pull/2914/commits/46fc1701e4b66dab7367d8a0face79cfa3b98cbd in this case).

Reply:
@jens-maus any thought on the rpi1 mod file? It came in with this commit https://github.com/home-assistant/operating-system/commit/9582183458570a93ddb4553cd1141ed8b7b7602b.

I am fine re-adding the two config symbols above, but I think we should cleanup the rpi-rf-mod package.

Reply:
Let's re-add them before merging and eventually address the rf-mod issues in another PR.